aten/src/ATen/DLConvertor.cpp

```diff
@@ -96,10 +96,14 @@ DLDataType getDLDataType(const Tensor& t) {
   return dtype;
 }
 
-static DLDevice getDLDevice(const Tensor& tensor, c10::DeviceIndex device_id) {
+DLDevice torchDeviceToDLDevice(at::Device device) {
   DLDevice ctx;
-  ctx.device_id = static_cast<int32_t>(static_cast<unsigned char>(device_id));
-  switch (tensor.device().type()) {
+
+  ctx.device_id = (device.is_cuda() || device.is_privateuseone())
```

Comment:
Why this new logic? We would we ignore the device index being passed in?

Reply:
I just got it from the place it was being called before. Needed it, since I'm also calling this function outside.

Reply:
Previously, this is how we were calling this function (the only call site):

https://github.com/pytorch/pytorch/blob/e1f28fe17bc55389fc5a31bad588d22e5ecca722/aten/src/ATen/DLConvertor.cpp#L317-L321

So, I basically, moved that code into this function. In other words, the logic did not change. The device index will always be either `src.get_device()`, i.e. `src.device().index()` (if the condition `src.is_cuda() || src.is_privateuseone()` is true), or 0 (otherwise).

Reply:
Ok, I guess it is another case where we should have an issue and tag other accelerator backends.